sync-core/src/main/java/com/cloudant/mazha/CouchClient.java

```diff
@@ -61,29 +59,57 @@ public URI getRootUri() {
         return this.uriHelper.getRootUri();
     }
 
-    HttpRequests getHttpClient() {
-        return httpClient;
-    }
-
-    public List<String> getList(URI uri) {
+    // TODO this exception handling code is a bit hairy
+    // it needs to catch eg FileNotFoundException and rethrow to emulate the previous exception handling behaviour
+    private InputStream executeToInputStream(HttpConnection connection) throws CouchException {
+        // TODO this header setting should go elsewhere
+        connection.requestProperties.put("Accept", "application/json");
         InputStream is = null;
+        String response = null;
+        int code = -1;
         try {
-            is = httpClient.get(uri);
-            return jsonHelper.fromJsonToList(new InputStreamReader(is), JSONHelper.STRING_LIST_TYPE_DEF);
-        } finally {
-            closeQuietly(is);
+            is = connection.executeToInputStream();
+        } catch (IOException ioe) {
+            System.out.println("original exception "+ioe);
+            try {
+                response = connection.getConnection().getResponseMessage();
+            } catch (IOException ioe2) {
+                // response can't be retrieved
+            }
         }
+        try {
+            // TODO restore this message in the last throw block
+            //String msg = String.format("Request: %s << Status: %s (%s) ", request.getRequestLine(), code, response.getStatusLine().getReasonPhrase()
+            code = connection.getConnection().getResponseCode();
+            if (code == 200 || code == 201 || code == 202) { // success (ok | created | accepted)
+                // ok
```

Comment:
A nice approach here is `code / 100 == 2` which covers us for any 200 code. Given we don't know whether any of the three we're checking for are strictly correct, we might as well go for any 200 code.